test/e2e/suite/nodeadm_test.go

```diff
@@ -220,109 +217,166 @@ var _ = Describe("Hybrid Nodes", func() {
-							Expect(os).NotTo(BeNil())
-							Expect(provider).NotTo(BeNil())
-
-							instanceName := test.instanceName("init", os, provider)
-
-							k8sVersion := test.cluster.KubernetesVersion
-							if test.overrideNodeK8sVersion != "" {
-								k8sVersion = suite.TestConfig.NodeK8sVersion
-							}
-
-							peeredNode := test.newPeeredNode()
-							instance, err := peeredNode.Create(ctx, &peered.NodeSpec{
-								InstanceName:   instanceName,
-								NodeK8sVersion: k8sVersion,
-								NodeNamePrefix: "simpleflow",
-								OS:             os,
-								Provider:       provider,
+					DescribeTableSubtree("Joining a node", Ordered,
+						func(os e2e.NodeadmOS, provider e2e.NodeadmCredentialsProvider) {
+							var instance ec2.Instance
+							var testCleanup TestCleanup
+
+							BeforeAll(func() {
+								Expect(os).NotTo(BeNil())
+								Expect(provider).NotTo(BeNil())
+								testCleanup.DeferAll()
+							})
+
+							It("creates EC2 instance", FlakeAttempts(2), func(ctx context.Context) {
+								instanceName := test.instanceName("init", os, provider)
+
+								k8sVersion := test.cluster.KubernetesVersion
+								if test.overrideNodeK8sVersion != "" {
+									k8sVersion = suite.TestConfig.NodeK8sVersion
+								}
+
+								peeredNode := test.newPeeredNode()
+								instanceNode, err := peeredNode.Create(ctx, &peered.NodeSpec{
+									InstanceName:   instanceName,
+									NodeK8sVersion: k8sVersion,
+									NodeNamePrefix: "simpleflow",
+									OS:             os,
+									Provider:       provider,
+								})
+								Expect(err).NotTo(HaveOccurred(), "EC2 Instance should have been created successfully")
+								testCleanup.Register(func(ctx context.Context) {
+									Expect(peeredNode.Cleanup(ctx, instanceNode)).To(Succeed())
+								})
+								Expect(ec2.WaitForEC2InstanceRunning(ctx, test.ec2Client, instanceNode.ID)).To(Succeed(), "EC2 Instance should have been reached Running status")
+								// Instance needed for future tests
+								instance = instanceNode
 							})
-							Expect(err).NotTo(HaveOccurred(), "EC2 Instance should have been created successfully")
-							DeferCleanup(func(ctx context.Context) {
-								Expect(peeredNode.Cleanup(ctx, instance)).To(Succeed())
-							}, NodeTimeout(deferCleanupTimeout))
 
-							test.logger.Info("Waiting for EC2 Instance to be Running...")
-							Expect(ec2.WaitForEC2InstanceRunning(ctx, test.ec2Client, instance.ID)).To(Succeed(), "EC2 Instance should have been reached Running status")
+							It("joins as a kubernetes nodes", func(ctx context.Context) {
+								node, err := kubernetes.WaitForNode(ctx, test.k8sClient, instance.IP, test.logger)
+								Expect(err).NotTo(HaveOccurred(), "Node should have joined the cluster")
+								Expect(node).NotTo(BeNil())
+							})
+
+							It("validates the kubernetes node", FlakeAttempts(2), func(ctx context.Context) {
```

Comment:
i would prefer not to repeat this
i understand the appeal but i fear it could hide flakiness in how we setup the node as opposed to flakiness in the validation